### PR TITLE
nvidia: switch to long-lived driver version 410.73

### PR DESCRIPTION
**Pull request by simonhdickson**

### CODE REVIEW

**Comment by Hoshpak** [comment]
Great work! Please change the title of the second commit, it should be "New package: nvidia390-390.87".

I will try to build the packages myself and review the content of the resulting binary packages. However I currently have no nvidia hardware installed in any of my systems that is supported by these driver versions. Did you test the multilib package, does everything work with 32bit apllication (e.g. 32 Windows games through wine that need direct3d). Did you test Vulkan with the new driver version?

**Comment by Hoshpak** [comment]
build on i686 fails: 

```
/usr/bin/strip:/destdir//nvidia-gtklibs-410.73/usr/lib/libnvidia-gtk3.so.410.73: File format not recognized
=> ERROR: nvidia-gtklibs-410.73_1: post-install_06-strip-and-debug-pkgs: '$STRIPCMD --strip-unneeded "$f"' exited with 1
=> ERROR:   in hook() at common/hooks/post-install/06-strip-and-debug-pkgs.sh:121
=> ERROR:   in run_func() at common/xbps-src/shutils/common.sh:21
=> ERROR:   in run_pkg_hooks() at common/xbps-src/shutils/common.sh:251
=> ERROR:   in main() at common/xbps-src/libexec/xbps-src-doinstall.sh:67
=> ERROR: nvidia-gtklibs-410.73_1: post-install_06-strip-and-debug-pkgs: 'find ${PKGDESTDIR} -type f' exited with 1
=> ERROR:   in hook() at common/hooks/post-install/06-strip-and-debug-pkgs.sh:59
=> ERROR:   in run_func() at common/xbps-src/shutils/common.sh:21
=> ERROR:   in run_pkg_hooks() at common/xbps-src/shutils/common.sh:251
=> ERROR:   in main() at common/xbps-src/libexec/xbps-src-doinstall.sh:67
```

It still packages the gtk libraries for x86_64.

File: srcpkgs/nvidia390/template [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+
+	# Omit drm dracut module too
+	vmkdir usr/lib/dracut/dracut.conf.d
+	echo "omit_dracutmodules+=\" drm \"" > ${DESTDIR}/usr/lib/dracut/dracut.conf.d/99-nvidia.conf
+
+	vmkdir /usr/share/X11/xorg.conf.d/
+	vinstall nvidia-drm-outputclass.conf 644 /usr/share/X11/xorg.conf.d 30-nvidia-drm-outputclass.conf
+}
+
+nvidia-gtklibs_package() {
```
Comment (Hoshpak) [inline srcpkgs/nvidia390/template]: This should match the source package name, `nvidia390-gtklibs`, same for the other sub-package functions.

**Comment by simonhdickson** [comment]
Ok I've pushed some changes that should fix that build, I'm not sure why it built for me. It was only copying some of the 64-bit binaries for 32bit now it should ignore them all instead. I'm just about to try out this new version.

Things I've tried:
- Max Payne (Windows 32bit DirectX)
- Doom (Windows 64bit Vulkan)
- Mad Max (Linux 64-bit Vulkan)
- Steam (Linux 32-bit)

**Comment by Hoshpak** [comment]
Now it fails while building `nvidia-dkms` on i686. Since i686 support seems to have ceased, we don't need that package here. I think it might be in order to conditionally use `subpackages` to only build the subpackages that are needed for i686/multilib (probably only `nvidia-libs`) and the rest only for `x86_64`.

**Comment by Hoshpak** [comment]
So I did some testing in a VM (without actual hardware present) now that the packages are building. While doing this, I found out that both versions are not compatible with linux4.19 but that is beyond the scope of this PR. I think the remaining issues here are:

* libnvidia-fatbinaryloader.so.390.87 needs to be changed to `nvidia390-libs` in `common/shlibs`
* at least `libnvidia-ml.so.1` also needs to be added to `common/shlibs` for `nvidia390-libs`

Before this can be merged, please squash the commits. There should be one for the nvidia changes and one for the introduction of nvidia390.

As the removal of the driver on i686 would leave people updating on i686 systems without a working graphics driver after an update, I think we should at least add an install message and point them to the nvidia390 driver as an alternative. This could be done conditionally as in the base-files INSTALL file, displaying the message via echo.

**Comment by simonhdickson** [comment]
I am getting a random but infrequent black screen on this driver, but I noticed this with the old driver too so it's not really a regression.

**Comment by Sir-Boops** [comment]
Oh, should've checked here before updating this myself :laughing: 

I've only been running it for a few hours but I haven't gotten any form of blank screen using this driver version! (Here's the template I updated with) -> https://gist.githubusercontent.com/Sir-Boops/cd99e72d12a4a217ca47397e730fef07/raw/f187d252198d96071489394301263d2238f2a868/gistfile1.txt

**Comment by simonhdickson** [comment]
I've been running this since Sunday, has been solid for me. I was comparing this package to the arch package the other day and I noticed we're not linking the new ray tracing libraries so I'll add those in later tonight.

**Pull request merged**